src/error.h

```diff
@@ -16,11 +16,21 @@ namespace lib
     {
         success = 0,
 
-        numberBytesLess20,
-        requestLengthShort,
+        numberOfBytesIsLessThan20,
+        requestLengthIsShort,
         eapMessageAttributeError
     };
 
+    class Exception: public std::runtime_error
+    {
+        public:
+            Exception(const boost::system::error_code& errorCode);
+            Error getErrorCode() const {return m_errorCode;}
+
+        private:
+            Error m_errorCode;
```

Comment:
Why do you use `Error` here if you get a `boost::system::error_code` and later convert the exception into the same `boost::system::error_code`?